app/assets/javascripts/hyrax/admin/admin_set_controls.es6

```diff
@@ -1,5 +1,29 @@
 export default class {
   constructor(elem) {
+    let url = window.location.pathname.replace('edit', 'files')
+    elem.find('#admin_set_thumbnail_id').select2({
+      ajax: { // Use the jQuery.ajax wrapper provided by Select2
+        url: url,
+        dataType: "json",
+        results: function(data, page) {
+          console.log(data)
+          return { results: data };
+        }
+      },
+      initSelection: function(element, callback) {
+          // the input tag has a value attribute preloaded that points to a preselected repository's id
+          // this function resolves that id attribute to an object that select2 can render
+          // using its formatResult renderer - that way the repository name is shown preselected
+          var id = $(element).val();
+          callback({ text: $(element).data('text') })
+          // if (id !== "") {
```

Comment:
Can you drop a comment here as to what the commented-out code is for (or remove it)?